language/diem-vm/src/diem_transaction_executor.rs

```diff
@@ -197,6 +197,54 @@ impl DiemVM {
         }
     }
 
+    fn execute_script_function<R: RemoteCache>(
```

Comment:
There's a fair bit of code duplication here. Would it work to share the code with execute_script? You could pass in the payload (for either Script or ScriptFunction) and match on it inside the function.

Reply:
I guess this is possible with a ScriptCodeOrFn enum. And then we can pass in the enum and match inside the execute_script function.

Reply:
I'm not sure it's worth adding a new enum. You can just pass the original payload, can't you? (and assert if the contents are anything other than a Script or a ScriptFunction) Either way works

Reply:
Ahh, yes, will update!

Reply:
Done